ui/helpers/utils/icon-factory.js

```diff
@@ -20,13 +20,13 @@ function IconFactory(jazzicon) {
 IconFactory.prototype.iconForAddress = function (
   address,
   diameter,
-  useStaticTokenList,
+  useTokenDetection,
   tokenList,
 ) {
-  // token from dynamic api list is fetched when useStaticTokenList is true
-  const addr = useStaticTokenList ? address : toChecksumHexAddress(address);
+  // token from dynamic api list is fetched when useTokenDetection is true
+  const addr = useTokenDetection ? address : toChecksumHexAddress(address);
```

Comment:
Same thing here. Maybe remove ternary and just toCheckSumHexAddress the address either way.

Reply:
When `useTokenDetection` flag is true, the `tokenList` will be fetched from the token list service API and the address of the token provided by the API is not a checksum address. Whereas when the flag is false the token list will be fetched from `contract-metadata` and then the address is a checksum address. So tokenList will hold checksum address and non-checksum address at different points based on the list from which it is fetched.

Reply:
Ok I still think we might as well checksum either way. It won't hurt if it's already checksummed? And if not doing that, could you add a comment explaining why we only checkSum the staticList addresses?

Reply:
Also here. I get why there's the ternary now. Please add a comment explaining?

Reply:
And here!